### PR TITLE
Set user location by IP

### PR DESCRIPTION
**Pull request by abaibik**
Add node_modules to gitignore, delete them from git

### CODE REVIEW

File: routes/update-routes.js [hunk 1]
```diff
@@ -123,6 +127,34 @@ module.exports = (app) => {
     }
   );
 
+  //This route sets geolocation of a user in the profile
+  app.put(
+    "/api/profile/location/geo",
+    passport.authenticate("jwt", { session: false }),
+    (req, res) => {
+      const userIP = req.headers["x-forwarded-for"] || req.ip;
```
Comment (NataliaDoudkina) [inline routes/update-routes.js:135]: Will it work in incognito mode?

File: routes/update-routes.js [hunk 2]
```diff
@@ -123,6 +127,34 @@ module.exports = (app) => {
...
+      const info = lookup.get(userIP);
+
+      if (!info || !info.country || !info.city || !info.location) {
```
Comment (NataliaDoudkina) [inline routes/update-routes.js:138]: why put check on info.country, ect  if there is a check if the whole info object exists? Like if info is null, info.country would not exist , or am I missing something ? lol

**Review by NataliaDoudkina — COMMENTED** [review]
Did you check how accurate are the coordinates?

**Comment by abaibik** [comment]
> Did you check how accurate the coordinates are? Is it free service? Will it work in mobile app?
It depends on a country: https://www.maxmind.com/en/geoip2-city-accuracy-comparison
Yes, it is free.
It is on a server side so yes :)

File: routes/update-routes.js (hunk 1 above)
Comment (abaibik) [reply routes/update-routes.js:135]: Yes

**Comment by NataliaDoudkina** [comment]
I tried to run it , but got info === null

**Review by NataliaDoudkina — APPROVED** [review]
Approved without testing on a real server

**Pull request merged**